humility-dump-agent/src/hiffy.rs

```diff
@@ -161,8 +170,12 @@ impl<'a> DumpAgent for HiffyDumpAgent<'a> {
 
         let results = self.run(ops.as_slice())?;
 
-        if let Err(err) = results[rindex] {
-            bail!("failed to take dump: {}", op.strerror(err));
+        match results[rindex] {
+            Err(IpcError::Error(err)) => {
+                bail!("failed to take dump: {}", op.strerror(err))
+            }
+            Err(e) => bail!("failed to take dump: {e}"),
```

Comment:
Same comment as above: printing the raw dead code is kinda cryptic here.

(also, I'd vote to spell out `Err(IpcError::ServerDied(..)) => ..` for future-proofing)

Reply:
The reason this doesn't do that is because `IdolOperation::strerror` lives in
the `humility-idol` crate, which is a dependency of `humility-hiffy`, which is
where `IpcError` is defined. Although, now that I've actually written that down,
there's no reason that `IpcError` has to be in `humility-hiffy`...

Reply:
Done!